lib/modulesync/repository.rb

```diff
@@ -51,17 +51,21 @@ def switch_branch(branch)
         repo.checkout("origin/#{branch}")
         repo.branch(branch).checkout
       else
-        repo.checkout('origin/master')
-        puts "Creating new branch #{branch}"
+        base_branch = default_branch
+        unless base_branch
+          puts "Couldn't detect default branch. Falling back to assuming 'master'"
+          base_branch = 'master'
+        end
+        puts "Creating new branch #{branch} from #{base_branch}"
+        repo.checkout("origin/#{base_branch}")
         repo.branch(branch).checkout
       end
     end
 
     def prepare_workspace(branch)
       # Repo needs to be cloned in the cwd
       if !Dir.exist?("#{@directory}/.git")
-        puts 'Cloning repository fresh'
-        puts "Cloning from '#{@remote}'"
+        puts "Cloning repository fresh from '#{@remote}'"
```

Comment:
Rubocop was annoyed the class was 1 too many lines, so I 'stole' back one here...